### PR TITLE
Add Openstack to supported providers

### PR DESCRIPTION
**Pull request by Maladie**
**Description**

Changes proposed in this pull request:

- Added possibility to create OpenStack cluster via graphql mutation
- diskType and volumeSizeGB fields are now optional (OpenStack requirement)
- Database schema adjusted accordingly
- Name parameter is now obligatory

**Related issue(s)**

### CODE REVIEW

File: components/provisioner/internal/model/infrastructure/openstack/cloud_profile.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CloudProfileConfig contains provider-specific configuration that is embedded into Gardener's `CloudProfile`
+// resource.
+type CloudProfileConfig struct {
```
Comment (franpog859) [inline components/provisioner/internal/model/infrastructure/openstack/cloud_profile.go:12]: Are the types from this file used anywhere?

File: components/provisioner/internal/model/gardener_config.go [hunk 2]
```diff
@@ -356,20 +314,82 @@ func (c AWSGardenerConfig) ExtendShootConfig(gardenerConfig GardenerConfig, shoo
...
-			VolumeSize: fmt.Sprintf("%dGi", gardenerConfig.VolumeSizeGB),
-		},
-		Maximum: int32(gardenerConfig.AutoScalerMax),
-		Minimum: int32(gardenerConfig.AutoScalerMin),
-		Zones:   zones,
+		Maximum:        int32(gardenerConfig.AutoScalerMax),
+		Minimum:        int32(gardenerConfig.AutoScalerMin),
+		Zones:          zones,
 	}
+
```
Comment (franpog859) [inline components/provisioner/internal/model/gardener_config.go:384]: I don't really remember where could it be but do you think it'll be useful to validate the graphql input with the optional `DiskType` and `VolumeSizeGB` fields? I assume that the provisioning and upgrading of the shoot will just fail if they regard the OpenStack clusters and these two fields are provided. I think we have someplace to validate the input so we could fail these requests super fast

File: components/provisioner/pkg/gqlschema/schema.graphql [hunk 3]
```diff
@@ -199,6 +207,13 @@ input AWSProviderConfigInput {
     internalCidr: String!   # Classless Inter-Domain Routing for the private subnet
 }
 
+input OpenStackProviderConfigInput {
+    zones:           [String!]! # Zones in which to create the cluster
+    floatingPoolName: String!  # FloatingPoolName name in which LoadBalancer FIPs should be created.
+    cloudProfileName: String!  # Name of the target Cloud Profile
+    loadBalancerProvider: String! # Name of load balancer provider, e.g. f5
```
Comment (franpog859) [inline components/provisioner/pkg/gqlschema/schema.graphql:214]:
```suggestion
    zones:           [String!]!   # Zones in which to create the cluster
    floatingPoolName: String!     # FloatingPoolName name in which LoadBalancer FIPs should be created.
    cloudProfileName: String!     # Name of the target Cloud Profile
    loadBalancerProvider: String! # Name of load balancer provider, e.g. f5
```

File: components/schema-migrator/migrations/provisioner/202119021531_drop_not_null_volume_size_gb_and_disk_type.down.sql [hunk 4]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE gardener_config ALTER COLUMN disk_type SET NOT NULL;
```
[the remark this answers is not in the document]
Comment (franpog859) [reply components/schema-migrator/migrations/provisioner/202119021531_drop_not_null_volume_size_gb_and_disk_type.down.sql:1]: It'll also good to test how it behaves when there the `NULL` values in the column already. Are these deleted then? I mean when migrating backwards with the `...down.sql` migration

[the remark this answers is not in the document]
Comment (Maladie) [reply components/schema-migrator/migrations/provisioner/202119021531_drop_not_null_volume_size_gb_and_disk_type.down.sql:1]: True, I will add update with default values to down.sql migration

File: components/provisioner/internal/model/gardener_config.go (hunk 2 above)
Comment (Maladie) [reply components/provisioner/internal/model/gardener_config.go:384]: Thanks!

File: components/schema-migrator/migrations/provisioner/202119021531_drop_not_null_volume_size_gb_and_disk_type.down.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (Maladie) [reply components/schema-migrator/migrations/provisioner/202119021531_drop_not_null_volume_size_gb_and_disk_type.down.sql:1]: Regarding transaction, I don't think it's necessary here. Those alterations are not closely related and don't really need to be performed together. However, it will be needed in down.sql when I add update with default values

File: components/provisioner/pkg/gqlschema/schema.graphql [hunk 5]
```diff
@@ -153,16 +160,16 @@ input ClusterConfigInput {
 }
 
 input GardenerConfigInput {
-    name: String                                    # Name of the cluster
+    name: String!                                   # Name of the cluster
```
[the remark this answers is not in the document]
Comment (Maladie) [reply components/provisioner/pkg/gqlschema/schema.graphql:163]: Ah, true! Added this to the description

File: components/provisioner/internal/model/infrastructure/openstack/cloud_profile.go (hunk 1 above)
Comment (Maladie) [reply components/provisioner/internal/model/infrastructure/openstack/cloud_profile.go:12]: All we need is LoadBalancerClass. The rest are just leftovers so I moved LoadBalancerClass to control_plane file and deleted cloud_profile. Thanks!

File: components/provisioner/internal/api/validator_test.go [hunk 6]
```diff
@@ -416,13 +415,13 @@ func initializeConfigs() (*gqlschema.ClusterConfigInput, *gqlschema.RuntimeInput
 	clusterConfig := &gqlschema.ClusterConfigInput{
 		GardenerConfig: &gqlschema.GardenerConfigInput{
```
[the remark this answers is not in the document]
Comment (Maladie) [reply components/provisioner/internal/api/validator_test.go:416]: Added

**Review by franpog859 — APPROVED** [review]
LGTM
